analytics/lib/counts.py

```diff
@@ -240,14 +274,11 @@ def do_pull_from_zerver(stat, start_time, end_time):
     FROM zerver_userprofile
     JOIN zerver_message
     ON
-    (
-        zerver_message.sender_id = zerver_userprofile.id AND
+        zerver_userprofile.id = zerver_message.sender_id AND
+        zerver_userprofile.date_joined < %%(time_end)s AND
         zerver_message.pub_date >= %%(time_start)s AND
         zerver_message.pub_date < %%(time_end)s
         %(join_args)s
-    )
-    WHERE
-            zerver_userprofile.date_joined < %%(time_end)s
```

Comment:
What's the thinking behind this part of the change?  It seems like some of these you kept the separate `WHERE` clause, and others you put it all into the `ON` clause...

AFAIK these do the same thing, and so one should do the semantically coherent thing, which I would usually expect to be the `ON` clause just having the parts actually involved in the join....